catroid/src/createatschool/java/org/catrobat/catroid/createatschool/ui/adapter/TemplateAdapter.java

```diff
@@ -110,14 +120,31 @@ public void onClick(View view) {
 	}
 
 	private void setImage(TemplateData templateData, ViewHolder holder) {
-		Picasso.with(context).load(templateData.templateResourceId).into(holder.image);
+		Picasso.with(context).load(getBaseUrl() + templateData.getThumbnail()).into(holder.image);
```

Comment:
Yes, it directly loads via http/https. If there's no Internet connection, this is handled by Picasso, so that no image is loaded at all (empty ImageView). However, this shouldn't happen as the Internet connection is already checked before templates are downloaded at all.